File: app/src/main/java/com/absinthe/anywhere_/ui/main/WebviewFragment.java
package com.absinthe.anywhere_.ui.main;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebSettings;
import android.webkit.WebView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.absinthe.anywhere_.R;
import com.absinthe.anywhere_.databinding.FragmentWebviewBinding;
import com.absinthe.anywhere_.ui.settings.SettingsActivity;
import com.absinthe.anywhere_.utils.manager.URLManager;

public class WebviewFragment extends Fragment {
    private static final String BUNDLE_URI = "BUNDLE_URI";

    private FragmentWebviewBinding mBinding;
    private String mUri;

    static WebviewFragment newInstance(String uri) {
        WebviewFragment fragment = new WebviewFragment();
        Bundle bundle = new Bundle();
        bundle.putString(BUNDLE_URI, uri);
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        mBinding = FragmentWebviewBinding.inflate(inflater, container, false);
        initView();
        return mBinding.getRoot();
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mUri = getArguments().getString(BUNDLE_URI);
        } else {
            mUri = URLManager.DOCUMENT_PAGE;
        }
    }

    @SuppressLint("SetJavaScriptEnabled")
    private void initView() {
        setHasOptionsMenu(true);
        WebView webView = mBinding.wvContainer;
        webView.setBackgroundColor(Color.TRANSPARENT);
        WebSettings webSettings = webView.getSettings();
        webSettings.setJavaScriptEnabled(true);
        webSettings.setAllowFileAccess(true);
        webSettings.setCacheMode(WebSettings.LOAD_NO_CACHE);
        webSettings.setDomStorageEnabled(true);
        webView.loadUrl(mUri);
    }

    @Override
    public void onCreateOptionsMenu(@NonNull Menu menu, @NonNull MenuInflater inflater) {
        inflater.inflate(R.menu.web_menu, menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == R.id.toolbar_settings) {
            startActivity(new Intent(getContext(), SettingsActivity.class));
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/absinthe/anywhere_/ui/main/WebviewFragment.kt
package com.absinthe.anywhere_.ui.main

import android.annotation.SuppressLint
import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.view.*
import android.webkit.WebSettings
import androidx.fragment.app.Fragment
import com.absinthe.anywhere_.R
import com.absinthe.anywhere_.databinding.FragmentWebviewBinding
import com.absinthe.anywhere_.ui.settings.SettingsActivity
import com.absinthe.anywhere_.utils.manager.URLManager

class WebviewFragment : Fragment() {

    private lateinit var mBinding: FragmentWebviewBinding
    private lateinit var mUri: String

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        mBinding = FragmentWebviewBinding.inflate(inflater, container, false)
        initView()
        return mBinding.root
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mUri = arguments?.getString(BUNDLE_URI) ?: URLManager.DOCUMENT_PAGE
    }

    @SuppressLint("SetJavaScriptEnabled")
    private fun initView() {
        setHasOptionsMenu(true)
        mBinding.wvContainer.apply {
            setBackgroundColor(Color.TRANSPARENT)

            settings.apply {
                javaScriptEnabled = true
                allowFileAccess = true
                domStorageEnabled = true
                cacheMode = WebSettings.LOAD_NO_CACHE
            }

            loadUrl(mUri)
        }
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        inflater.inflate(R.menu.web_menu, menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.toolbar_settings) {
            startActivity(Intent(context, SettingsActivity::class.java))
        }
        return super.onOptionsItemSelected(item)
    }

    companion object {
        private const val BUNDLE_URI = "BUNDLE_URI"

        fun newInstance(uri: String?): WebviewFragment {
            return WebviewFragment().apply {
                arguments = Bundle().apply {
                    putString(BUNDLE_URI, uri)
                }
            }
        }
    }
}